src/main/java/org/opensearch/knn/index/vectorvalues/KNNMergeVectorValues.java

```diff
@@ -0,0 +1,113 @@
+                    final Bits liveDocs = mergeState.liveDocs[i];
+                    final int liveDocsInt;
+                    if (liveDocs != null) {
+                        if (liveDocs instanceof FixedBitSet) {
+                            liveDocsInt = ((FixedBitSet) liveDocs).cardinality();
+                        } else {
+                            liveDocsInt = computeLiveDocs(values, liveDocs);
+                            values = knnVectorsReader.getFloatVectorValues(fieldInfo.name);
+                        }
+                    } else {
+                        liveDocsInt = Math.toIntExact(values.cost());
+                    }
+                    subs.add(new KNNVectorValuesSub<>(mergeState.docMaps[i], values, liveDocsInt));
+                }
+            }
+        }
+        return subs;
+    }
+
+    public static List<KNNVectorValuesSub<ByteVectorValues>> mergeByteVectorValues(FieldInfo fieldInfo, MergeState mergeState)
+        throws IOException {
+        assert fieldInfo != null && fieldInfo.hasVectorValues();
+        if (fieldInfo.getVectorEncoding() != VectorEncoding.BYTE) {
+            throw new UnsupportedOperationException("Cannot merge vectors encoded as [" + fieldInfo.getVectorEncoding() + "] as BYTE");
+        }
+        final List<KNNVectorValuesSub<ByteVectorValues>> subs = new ArrayList<>();
+        for (int i = 0; i < mergeState.knnVectorsReaders.length; i++) {
+            KnnVectorsReader knnVectorsReader = mergeState.knnVectorsReaders[i];
+            if (knnVectorsReader != null) {
+                ByteVectorValues values = knnVectorsReader.getByteVectorValues(fieldInfo.name);
+                if (values != null) {
+                    final Bits liveDocs = mergeState.liveDocs[i];
+                    final int liveDocsInt;
+                    if (liveDocs != null) {
+                        if (liveDocs instanceof FixedBitSet) {
+                            liveDocsInt = ((FixedBitSet) liveDocs).cardinality();
+                        } else {
+                            liveDocsInt = computeLiveDocs(values, liveDocs);
+                            values = knnVectorsReader.getByteVectorValues(fieldInfo.name);
+                        }
+                    } else {
+                        liveDocsInt = Math.toIntExact(values.cost());
+                    }
+                    subs.add(new KNNVectorValuesSub<>(mergeState.docMaps[i], values, liveDocsInt));
+                }
+            }
+        }
+        return subs;
+    }
+
+    private static int computeLiveDocs(final DocIdSetIterator values, Bits liveDocs) throws IOException {
+        int count = 0;
+        if (liveDocs != null) {
+            while (values.docID() != DocIdSetIterator.NO_MORE_DOCS) {
+                count++;
+                values.nextDoc();
+            }
+        }
+        return count;
+    }
+
+    static class KNNVectorValuesSub<T extends DocIdSetIterator> extends DocIDMerger.Sub {
```

Comment:
why this is package private?

Reply:
Needed in [KNNVectorValuesIterator](https://github.com/opensearch-project/k-NN/pull/2135/files#diff-5b15a95bce01061c109452cff568a77fbef8d9a79f5efe7c3161b5d745c567d0R194)